### PR TITLE
Models and validations

### LINKED ISSUE #7: [1.5xpt]Create the Models, their associations and validation
- [ ] Create the validations
- [ ] Create the Assosiations
- [ ] User Model
- [ ] Reservations Model
- [ ] Motors

### PR DESCRIPTION
**Pull request by NatanimA**
### In this PR

- [ ] Inserted Models such as, (User, Motorcycle, and Reservation)
- [ ] Inserted Validations for all the models

### CODE REVIEW

File: app/models/motorcycle.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Motorcycle < ApplicationRecord
+  belongs_to :user, class_name: 'User'
+  has_many :reservations, foreign_key: 'motorcycle_id'
+
+  validates :name, presence: true, length: { minimum: 3, maximum: 50 }
+  validates :img_url, presence: true
+  validates :price, presence: true, numericality: { greater_than: 0 }
```
Comment (karayamanemre) [inline app/models/motorcycle.rb:7]: There are a few more columns in the motorcycle table to be validated. Such as 'description', 'model_year', 'engine' and 'fuel_type'.

File: app/models/motorcycle.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Motorcycle < ApplicationRecord
+  belongs_to :user, class_name: 'User'
```
Comment (karayamanemre) [inline app/models/motorcycle.rb]: I think motorcycles can have many user to reserve them. You can change this line to:
`has_many :users, through: :reservations`

**Review by karayamanemre — CHANGES_REQUESTED** [review]
Hi @NatanimA 
Thanks for creating the required models! Great job! 👍🏽 
However, I noticed some changes required. Please check them.

File: app/models/motorcycle.rb (hunk 2 above)
Comment (NatanimA) [reply app/models/motorcycle.rb]: @karayamanemre  Are we going to refer how many users a motorcyle have?

File: app/models/motorcycle.rb (hunk 1 above)
Comment (NatanimA) [reply app/models/motorcycle.rb:7]: @karayamanemre  Noted

File: app/models/motorcycle.rb (hunk 2 above)
Comment (karayamanemre) [reply app/models/motorcycle.rb]: I don't think we need to refer. @NatanimA

**Review by karayamanemre — APPROVED** [review]
Hi @NatanimA 
Thanks for making the changes to the Motorcycle model! Nice job! 💯

**Pull request merged**